torchqdynamics/operators.py

```diff
@@ -0,0 +1,216 @@
+) -> Tensor | tuple[Tensor, ...]:
+    """Bosonic creation operator.
+
+    If only a single dimension is provided, `create` returns the creation operator of
+    corresponding dimension. If instead multiples dimensions are provided, `create`
+    returns a tuple of each creation operator of given dimension, in the Hilbert space
+    given by the product of all dimensions.
+
+    Example:
+        >>> tq.create(4)
+        tensor([[0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j],
+                [1.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 1.414+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 0.000+0.j, 1.732+0.j, 0.000+0.j]],
+                dtype=torch.complex128)
+        >>> a, b = tq.create(2, 3)
+        >>> a
+        tensor([[0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j],
+                [0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j],
+                [0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j],
+                [1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j],
+                [0.+0.j, 1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j],
+                [0.+0.j, 0.+0.j, 1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j]],
+                dtype=torch.complex128)
+        >>> b
+        tensor([[0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j],
+                [1.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 1.414+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 0.000+0.j, 0.000+0.j, 1.000+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j, 1.414+0.j, 0.000+0.j]],
+                dtype=torch.complex128)
+    """
+    dims = _extract_tuple_from_varargs(dims)
+
+    # compute first destroy operator
+    adag = _create_single(dims[0], dtype=dtype, device=device)
+    if len(dims) == 1:
+        return adag
+
+    # compute all creation operators
+    ops = [adag]
+    eye = qeye(dims[0], dtype=dtype, device=device)
+    for i, dim in enumerate(dims[1:]):
+        # single mode operators
+        _adag = _create_single(dim, dtype=dtype, device=device)
+        _eye = qeye(dim, dtype=dtype, device=device)
+
+        # update ops
+        ops.append(tensprod(eye, _adag))
+        for j in range(i + 1):
+            ops[j] = tensprod(ops[j], _eye)
+
+        # update eye
+        eye = tensprod(eye, _eye)
+
+    return tuple(ops)
+
+
+def _create_single(
+    dim: int, *, dtype: dtype = torch.complex128, device: device | None = None
+) -> Tensor:
+    """Bosonic creation operator of a single mode."""
+    return torch.diag(torch.sqrt(torch.arange(1, dim, device=device)), -1).to(dtype)
+
+
+def displace(
+    dim: int,
+    alpha: complex,
+    *,
+    dtype: dtype = torch.complex128,
+    device: device | None = None,
+) -> Tensor:
+    """Single-mode displacement operator."""
```

Comment:
```suggestion
    """Displacement operator."""
```
I think it's always single mode by default, no need to precise it everywhere.